### PR TITLE
feat: #579 - new method to log in, that returns data about user

### LINKED ISSUE #579: New login method
### Why - Problem description
cf. https://github.com/openfoodfacts/smooth-app/issues/3118#issuecomment-1274220392
Code evolution on the server side: now the login (`'/cgi/auth.pl'`) may return a JSON (an interesting one btw) if successful, instead of an empty body. Problem: we used to say "if the body is not empty, it's not good it's an error"

### What - Proposed solution
As @stephanegigandet suggested, the current "empty" algorithm still works but will soon be deprecated.
The new solution is:
* add `?body=1` to the URL
* check the body, and its possible typical values
  * `{"status":0,"status_verbose":"user not signed-in"}` (wrong user and/or password)
  * `{"status":1,"status_verbose":"user signed-in","user":{"email":"blababla@gmail.com","name":"Mr. John Doe"},"user_id":"gqwbgsvvod"}` (success)
  * this way we can even retrieve the email, the name and the user id (before, we got nothing, and even a possible confusion between email and user id)

### PR DESCRIPTION
**Pull request by monsieurtanuki**
New file:
* `LoginStatus.dart`: Status after an attempt to log in.

Impacted files:
* `api_getProduct_test.dart`: unrelated typo fix
* `openfoodfacts.dart`: new method `login2` that returns a `LoginStatus`
* `user_management_test_prod.dart`: now uses `login2` instead of `login`
* `user_management_test_test_env.dart`: now uses `login2` instead of `login`

### What
- New method `login2` that returns data about user

### Fixes bug(s)
- Closes #579

### CODE REVIEW

File: lib/model/LoginStatus.dart [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+  final int status;
+  final String statusVerbose;
+  final String? userEmail;
+  final String? userName;
+  final String? userId;
+
+  factory LoginStatus.fromJson(Map<String, dynamic> json) => LoginStatus(
+        status: JsonObject.parseInt(json['status'])!,
+        statusVerbose: json['status_verbose'] as String,
```
Comment (M123-dev) [inline lib/model/LoginStatus.dart:38]: We were talking about not failing if some parameter is missing, we could make this nullable but to be honest it's not that important here

File: lib/model/LoginStatus.dart [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        userId: json['user_id'] as String?,
+        userEmail: json['user']['email'] as String?,
+        userName: json['user']['name'] as String?,
+      );
+
+  /// Was the login successful?
+  bool get ok => status == 1;
```
Comment (M123-dev) [inline lib/model/LoginStatus.dart:45]: Ok is si generic

I would prefer loggedIn or correctCredentials or something like that

**Review by M123-dev — APPROVED** [review]
Perfect @monsieurtanuki looks good to me, I was thinking about how to name the method while not removing the old one. Login2 is perfectly fine when we switch when all users should have migrated.


Just one suggestion to keep the OpenFoodApiClient (even if just a little shorter) we can make `login` call and return `login2.ok`

**Comment by monsieurtanuki** [comment]
Thank you @M123-dev for your review! I applied some of your suggestions.
I'll merge this PR when the tests are OK, or within 2 hours, whichever comes first ;)

**Comment by monsieurtanuki** [comment]
@M123-dev Feel free to merge 1.26.0.

**Comment by monsieurtanuki** [comment]
@M123-dev But you'd better be ready for Smoothie and remove the possible warnings (e.g. deprecation) the new release creates.

**Pull request merged**